homeassistant/components/bmw_connected_drive/config_flow.py

```diff
@@ -41,8 +41,13 @@ async def validate_input(
 
     try:
         await auth.login()
-    except HTTPError as ex:
-        raise CannotConnect from ex
+    except (HTTPError, HTTPStatusError) as err:
+        if isinstance(err, HTTPStatusError) and err.response.status_code in (
+            401,
+            403,
+        ):
+            raise InvalidAuth from err
```

Comment:
I'm general. These type of exceptions makes sense to wrap in the lib. So the lib doesn't exist the http library its using.

Reply:
Makes sense! Should I add exception handy to the lib an bump the dependency on this PR?

Reply:
Yes i think that would be good.

Reply:
Could you maybe have a look at https://github.com/bimmerconnected/bimmer_connected/pull/522?
Basically, the library would raise only `MyBMWAuthException` or `MyBMWAPIException` which we then could catch here.